### ISSUE TITLE
noscript + firefox containers

### ISSUE THREAD

**Issue by mfr-itr** [hackademix/noscript#5]
Is there a way to block a domain in all tabs except one specific container [Firefox container](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/)? That way, I could have a Youtube container with google scripts activated, and finally ditch chromium.

**Comment by jameysharp** [comment]
I came here to suggest this too. For example, I'd like to allow scripts from twitter.com but only when I'm in the permanent container I use for accessing Twitter.

Although I suppose if I could say "*.twitter.com and *.twimg.com should be Trusted but only when the tab URL is at *.twitter.com", that would do the same thing for every use case I can think of, without requiring noscript to know anything about containers. @mfr-itr, what do you think of that alternative?

**Comment by mfr-itr** [comment]
Hum, did not know that was possible. At first glance it seems good enough, not sure if it would be sufficient though...
Edit: my bad, I thought you meant it was possible right now.

**Comment by ThrawnCA** [comment]
I'm pretty sure that that kind of thing is technically possible under the hood, although the UI might not fully support it.

**Comment by rastapasta** [comment]
I would *love* to see container support similar to Cookie AutoDelete's one - it currently allows to setup cookie retention policies based on the used Firefox container. Implemented pretty straight forward by selecting different configuration stores based on the tab's container context, [here](https://github.com/Cookie-AutoDelete/Cookie-AutoDelete/blob/470d00048c20e69ec4bf37074a25e9507c482d92/src/redux/Actions.ts#L219-L256) on GitHub.

**Comment by dnut** [comment]
> although the UI might not fully support it.

Why would this require UI changes?

**Comment by leonixyz** [comment]
I'd be happy if all site permissions were "namespaced" within Firefox containers. Each container can have a different set of settings, and there is IMO no need to change the UI.

**Comment by oliversturm** [comment]
Any chance somebody is working on this?

**Comment by Wuestenschiff** [comment]
Yes please this would be awsome. Cookie Autodelete already has this feature

**Comment by jtotht** [comment]
There’s a [contextualIdentities API]( https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/contextualIdentities), but I don’t find any way to query the container of a particular tab. Even worse, if NoScript asks for permission to access containers, Firefox will enable containers even if they would be disabled otherwise. This forces users to use a feature that they don’t need, which is bad user experience (mysterious menu options, maybe also a small but non-zero runtime cost). Probably NoScript is popular enough to convince Mozilla to provide a solution that covers our use case?
